### ISSUE TITLE
Tokio 1.x runtime error while upgrading from 0.11.1 to 0.12.1

### ISSUE THREAD

**Issue by suvl** [timberio/vector#6887]
While trying to upgrade some kubernetes workloads (shipping logs from kafka to GCS, if that detail matters) I'm getting a Tokio runtime error:

```
thread 'main' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.2.0/src/runtime/blocking/pool.rs:85:33
```

Full log gist [here](https://gist.github.com/suvl/5663cc08665f33957303c21ae5134a2b).

I tried both alpine and debian images, and tried to remove my CLI parameters (that are ` --threads=2 --watch-config=true --require-healthy=false`) to no avail. Let me know if the config file is relevant for this case. Thanks!

**Comment by jszwedko** [comment]
Hi @suvl . Thanks for reporting this! Are you able to share the config you are using too?

**Comment by suvl** [comment]
@jszwedko sure, [here it is](https://gist.github.com/suvl/481e3109619608d508b1efa8b9ba8f37).

**Comment by jszwedko** [comment]
Thanks for this report @suvl . I realized it seems to be the same as #6712 and should be fixed by #6574 . We'll releasing a patch release shortly with this change.

**Comment by jszwedko** [comment]
Closing this out since https://github.com/timberio/vector/pull/6574 was merged, but please reopen if you still see this issue as part of the 0.12.2 release (building now).
